types/tx/signing/signature.go

```diff
@@ -84,3 +85,29 @@ func SignatureDataFromProto(descData *SignatureDescriptor_Data) SignatureData {
 		panic(fmt.Errorf("unexpected case %+v", descData))
 	}
 }
+
+var _, _ codectypes.UnpackInterfacesMessage = &SignatureDescriptors{}, &SignatureDescriptor{}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (sds *SignatureDescriptors) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	for _, sig := range sds.Signatures {
+		err := sig.UnpackInterfaces(unpacker)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// UnpackInterfaces implements the UnpackInterfaceMessages.UnpackInterfaces method
+func (sd *SignatureDescriptor) UnpackInterfaces(unpacker codectypes.AnyUnpacker) error {
+	var pubKey crypto.PubKey
+	err := unpacker.UnpackAny(sd.PublicKey, &pubKey)
```

Comment:
why do we need to provide the second argument here? Can't the `unpacker` deduce they type based on the first argument?

Reply:
Hmm, how? The first argument is a `Any`

Reply:
Ah, OK. I thought it's `crypto.PublicKey`

Reply:
Got it, that's probably on master. This PR is changing this field from `crypto.PublicKey` to `Any`.